### DISCUSSION TITLE
Instruções de uso com bitbucket

### DISCUSSION THREAD

**Discussion by eduardomrj** [HashLoad/boss#92] [Q&A] [answered]
Olá pessoal!

Existe algum documento informando como usar o BOSS com bitbucket? 
Não consigo usar por conta do certificado e não achei documentação.

**Comment by andre-djsystem** [comment, marked as answer]
Tente utilizar o comando boss login

**Reply by viniciussanchez** [reply]
Isso mesmo, tem que fazer o boss login, e se possível usar a opção com SSH

**Reply by eduardomrj** [reply]
Não funciona...
Segui todos os passo do video : https://www.youtube.com/watch?v=geAGT-M0RGY
Fiz login no boss

Retorna o erro: 
[ERROR] Error to get repository of github.com/git@bitbucket.org:automacaocom/test001: unexpected client error: unexpected requesting "https://github.com/git@bitbucket.org:automacaocom/test001/info/refs?service=git-upload-pack" status code: 400

OBS:
Veja que fiz o login e configuração no servidor do bitbucket mas no erro está dando um erro no GITHUB.com ... eu não tenho nem conta no GITHUB

**Reply by snakeice** [reply]
Poderia passar o comando completo? Do install

**Reply by eduardomrj** [reply]
C:\test>boss install git@bitbucket.org:automacaocom/test001.git
[INFO ] Installing modules in project path
[INFO ] Building cache files...
[INFO ] Updating cache of dependency test001
[INFO ] Downloading dependency github.com/git@bitbucket.org:automacaocom/test001
[ERROR] Error to get repository of github.com/git@bitbucket.org:automacaocom/test001: unexpected client error: unexpected requesting "https://github.com/git@bitbucket.org:automacaocom/test001/info/refs?service=git-upload-pack" status code: 400

**Reply by eduardomrj** [reply]
arquivo de configuração
{
	"id": "92c6ee1f5ad7d4339de3bc242df356f3",
	"auth": {
		"bitbucket.org": {
			"use": true,
			"path": "c:\\ssh\\bitbucket-lib"
		}
	},
	"purge_after": 3,
	"internal_refresh_rate": 5,
	"last_purge_cache": "2022-06-01T21:05:56.3951037-03:00",
	"last_internal_update": "2022-06-01T20:49:57.0257955-03:00",
	"delphi_path": "C:\\Program Files (x86)\\Embarcadero\\Studio\\21.0\\bin",
	"config_version": 6,
	"git_embedded": true
}

**Reply by snakeice** [reply]
tenta `boss install bitbucket.org/automacaocom/test001`

**Reply by eduardomrj** [reply]
acho que agora vai! rsrsrs   mas deu um erro por que coloquei senha na chave privada, tenho que remover a senha ?

[ERROR] Fail to decrypt PassPhrase.
%!(EXTRA *errors.errorString=cipher text block size is too short)panic: ssh: this private key is passphrase protected

**Reply by andre-djsystem** [reply]
Qual versão do boss vc está usando? E no comando boss login vc informou o passphrase?

**Reply by eduardomrj** [reply]
Versão 3.0.7

Corrigi e coloquei a senha, arquivo de .cfg atualizado com pass.

Agora outro erro.

C:\libdelphi>boss install bitbucket.org/git@bitbucket.org:automacaocom/test001.git
[INFO ] Installing modules in project path
[INFO ] Building cache files...
[INFO ] Updating cache of dependency test001
[INFO ] Downloading dependency bitbucket.org/git@bitbucket.org:automacaocom/test001
[ERROR] Error to get repository of bitbucket.org/git@bitbucket.org:automacaocom/test001: unable to find any valid known_hosts file, set SSH_KNOWN_HOSTS env variable

**Reply by andre-djsystem** [reply]
Como o @snakeice citou acima, use  boss install bitbucket.org/automacaocom/test001

**Reply by eduardomrj** [reply]
**boss install bitbucket.org/automacaocom/test001**
[INFO ] Installing modules in project path
[INFO ] Building cache files...
[INFO ] Updating cache of dependency test001
[INFO ] Downloading dependency bitbucket.org/automacaocom/test001
[ERROR] Error to get repository of bitbucket.org/automacaocom/test001: unable to find any valid known_hosts file, set SSH_KNOWN_HOSTS env variable

**Reply by andre-djsystem** [reply]
https://docs.gitlab.com/ee/ci/ssh_keys/#verifying-the-ssh-host-keys

**Reply by eduardomrj** [reply]
Deu certo pessoal! obrigado a todos que ajudaram, sei que o tempo de vocês é preciso.

Vou deixar o registro de como foi resolvido.

- boss atualizado com ultima versão
- chave ssh criada com senha
- chave ssh PUBLICA adicionada ao repositório bitbucket
- boss LOGIN apontando para a chave ssh e informando a senha (importante)

PROBLEMA : unable to find any valid known_hosts file
Resolução
- #Windows:
ssh-keyscan -t rsa bitbucket.org >> %userprofile%/.ssh/known_hosts

#Mac:
ssh-keyscan -t rsa bitbucket.org >> ~/.ssh/known_hosts

- depois boss install e tudo pronto!

Mais uma vez grato ao time pela ajuda fantástica!
